src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/DateFilter.php

```diff
@@ -0,0 +1,245 @@
+                $clause = [
+                    'range' => [
+                        $attributePath => [
+                            'lt' => $this->getFormattedDate($attributeCode, $value),
+                        ]
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addFilter($clause);
+
+                break;
+            case Operators::GREATER_THAN:
+                $clause = [
+                    'range' => [
+                        $attributePath => [
+                            'gt' => $this->getFormattedDate($attributeCode, $value),
+                        ]
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addFilter($clause);
+
+                break;
+            case Operators::BETWEEN:
+                $values = array_values($value);
+                $clause = [
+                    'range' => [
+                        $attributePath => [
+                            'gte' => $this->getFormattedDate($attributeCode, $values[0]),
+                            'lte' => $this->getFormattedDate($attributeCode, $values[1]),
+                        ]
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addFilter($clause);
+
+                break;
+            case Operators::NOT_BETWEEN:
+                $values = array_values($value);
+                $betweenClause = [
+                    'range' => [
+                        $attributePath => [
+                            'gte' => $this->getFormattedDate($attributeCode, $values[0]),
+                            'lte' => $this->getFormattedDate($attributeCode, $values[1]),
+                        ]
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addMustNot($betweenClause);
+                $this->searchQueryBuilder->addFilter($this->getExistsClause($attributePath));
+
+                break;
+            case Operators::IS_EMPTY:
+                $this->searchQueryBuilder->addMustNot($this->getExistsClause($attributePath));
+
+                break;
+            case Operators::IS_NOT_EMPTY:
+                $this->searchQueryBuilder->addFilter($this->getExistsClause($attributePath));
+
+                break;
+            case Operators::NOT_EQUAL:
+                $mustNotClause = [
+                    'term' => [
+                        $attributePath => $this->getFormattedDate($attributeCode, $value)
+                    ]
+                ];
+
+                $this->searchQueryBuilder->addMustNot($mustNotClause);
+                $this->searchQueryBuilder->addFilter($this->getExistsClause($attributePath));
+
+                break;
+            default:
+                throw InvalidOperatorException::notSupported($operator, static::class);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param string $field
+     *
+     * @return array
+     */
+    protected function getExistsClause($field)
```

Comment:
I really find this function useless, not relevant, and imo it makes the readability of your code much harder :/